src/main/scala/co/teapot/tempest/server/TempestDBServer.scala

```diff
@@ -75,85 +66,67 @@ class TempestDBServer(databaseClient: TempestDatabaseClient, config: TempestDBSe
   }
 
   def kStepNeighborsFiltered(edgeType: String,
-                             sourceId: Long,
+                             source: Node,
```

Comment:
I'm now conflicted on this.  The main drawback is that we double the number of methods in TempestDBServer.  Some methods decompose naturally into ThriftNode => Node, function(Nodes), Node => ThriftNode. But others, like outNeighbors, now use helper functions that change that code-flow, so I don't think I would re-use code if I had an outNeighbors call on ThriftNodes and a separate one on Nodes.  

I think if we really cared about efficiency, we'd want to back to using Ints (as opposed to Node or ThriftNode), but for a call like connectedComponent, it's awkward to not have typed nodes.  I think we should hold off on this until we better understand the efficiency concerns.  TempestDBServer has too many methods as it is.